### DISCUSSION TITLE
[10.x] Custom pivot table uses the wrong table name

### DISCUSSION THREAD

**Discussion by MR-AMDEV** [laravel/framework#48284] [Q&A]
### Laravel Version

10.20.0

### PHP Version

8.2.6

### Database Driver & Version

8.0.31

### Description

I am trying to use a custom pivot table to define a cast. The custom pivot table uses the default laravel naming conventions for the database table while making a query.

**Actual Result:**
The database error i got:
>SQLSTATE[42S02]: Base table or view not found: 1146 Table 'my_database.extra_inquiry' doesn't exist

Even if i specifically define the `protected $table = 'inquiry_extras';` on the InquiryExtra table it doesn't respect that.

**Expected Result:**
It should be using the table name **"inquiry_extras"** (as thats what we expect to get from **InquieryExtra** model) so the database donot throw the table not found error.

```
class Inquiry extends Model
{
    public function extras()
    {
        return $this->belongsToMany(Extra::class)
                    ->using(InquiryExtra::class)
                    ->withPivot(['quantity', 'price', 'total_price'])
                    ->withTimestamps();
    }
}
```

```
class InquiryExtra extends Pivot
{
    protected $casts = [
        'price' => 'float',
        'total_price' => 'float',
    ];
}
```

**Comment by crynobone** [comment]
Looking at the code you should be able to do the following:

```php
return $this->belongsToMany(Extra::class, InquiryExtra::class);
```

**Reply by MR-AMDEV** [reply]
Doing that uses the correct table name but do not cast the pivot columns. Already tried that.

**Comment by MR-AMDEV** [comment]
Do we have any other solution to handle casting on the pivot model?
